### PR TITLE
Migrated test_peer_detach_check_warning_message

### PR DESCRIPTION
**Pull request by aujjwal-redhat**
Migrated the [test
case](https://github.com/gluster/glusto-tests/blob/HEAD/tests/functional/glusterd/test_peer_detach_check_warning_message.py)

Updates: #292

Fixes: #356

Signed-off-by: Ayush Ujjwal <aujjwal@redhat.com>

### CODE REVIEW

**Comment by aujjwal-redhat** [comment]
```js
Summary:
+---------------+-------+--------------+
|    Category   | Cases | Pass Percent |
+---------------+-------+--------------+
| nonDisruptive |   0   |      0       |
|   Disruptive  |   1   |    100.0     |
|     Total     |   1   |    100.0     |
+---------------+-------+--------------+
Table:
 test_peer_detach_check_warning_message
 
+-------------+-------------+--------------------+
| Volume Type | Test Result |  Time taken (sec)  |
+-------------+-------------+--------------------+
|   Generic   |     PASS    | 14.718845844268799 |
+-------------+-------------+--------------------+
```


## LOGS:
```js
[2021-06-07 16:03:08,168] INFO  - Running gluster pool list --xml on 10.70.43.74
[2021-06-07 16:03:08,393] DEBUG  - {'msg': {'opRet': '0', 'opErrno': '0', 'opErrstr': None, 'peerStatus': {'peer': [{'uuid': '46b98e38-da19-423a-a055-76365c5308eb', 'hostname': 'serverc', 'hostnames': {'hostname': 'serverc'}, 'connected': '1', 'state': '3', 'stateStr': 'Peer in Cluster'}, {'uuid': '0bd0d62f-6f95-477f-9106-cd1c620def21', 'hostname': '10.70.43.115', 'hostnames': {'hostname': '10.70.43.115'}, 'connected': '1', 'state': '3', 'stateStr': 'Peer in Cluster'}, {'uuid': 'adf863d7-9a2a-45c3-86a9-4f520c7e7092', 'hostname': 'localhost', 'connected': '1'}]}}, 'Flag': True, 'node': '10.70.43.74', 'cmd': 'gluster pool list --xml', 'error_code': 0}
[2021-06-07 16:03:08,617] DEBUG  - {'msg': ['All clients mounted through the peer which is getting detached need to be remounted using one of the other active peers in the trusted storage pool to ensure client gets notification on any changes done on the gluster configuration and if the same has been done do you want to proceed? (y/n) '], 'Flag': True, 'node': '10.70.43.74', 'cmd': 'gluster peer detach 10.70.43.115', 'error_code': 0}
[2021-06-07 16:03:08,618] INFO  - Running gluster --xml peer status on 10.70.43.74
```

**Comment by srijan-sivakumar** [comment]
@aujjwal-redhat will check the test once in CI run..

**Comment by aujjwal-redhat** [comment]
> @aujjwal-redhat will check the test once in CI run..

Ack

File: common/ops/gluster_ops/peer_ops.py [hunk 1]
```diff
@@ -378,6 +378,19 @@ def is_peer_connected(self, servers: list, node: str) -> bool:
                                   f"not in connected state")
                 is_connected = False
 
+        # check which server in servers is not part of the pool itself
```
Comment (nik-redhat) [inline common/ops/gluster_ops/peer_ops.py:381]: Do add check above this to return if `is_connected` is false, we can avoid the later part of the execution.

**Comment by aujjwal-redhat** [comment]
```js
Table:
 test_peer_detach_check_warning_message
 
+-------------+-------------+-------------------+
| Volume Type | Test Result |  Time taken (sec) |
+-------------+-------------+-------------------+
|   Generic   |     PASS    | 14.30363392829895 |
+-------------+-------------+-------------------+



Framework runtime : 23.305354595184326
```

**Review by srijan-sivakumar — APPROVED** [review]
LGTM

**Comment by nik-redhat** [comment]
@srijan-sivakumar have you completed testing in the CI so that we can merge this?

**Comment by srijan-sivakumar** [comment]
@nik-redhat forgot to merge it..

**Pull request merged**